src/Swagger/Extractor/CollectionGetExtractor.php

```diff
@@ -0,0 +1,157 @@
+     */
+    public function __construct(
+        ResourceMetadataFactoryInterface $resourceMetadataFactory,
+        SwaggerDefinitions $swaggerDefinitions,
+        SwaggerFilterDefinitions $swaggerFilterDefinitions,
+        SwaggerExtractorInterface $decoratedExtractor = null
+    ) {
+        $this->decoratedExtractor = $decoratedExtractor;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->swaggerDefinitions = $swaggerDefinitions;
+        $this->swaggerFilterDefinitions = $swaggerFilterDefinitions;
+    }
+
+    public function extract(SwaggerMetadata $swaggerMetadata): SwaggerDocumentation
+    {
+        if (null !== $this->decoratedExtractor) {
+            $documentation = $this->decoratedExtractor->extract($swaggerMetadata);
+        } else {
+            $documentation = new SwaggerDocumentation();
+        }
+
+        if ($swaggerMetadata->isCollection() && in_array(Request::METHOD_GET, $swaggerMetadata->getMethods())) {
+            $resourceMetadata = $this->resourceMetadataFactory->create($swaggerMetadata->getResourceClass());
+
+            $this->extractSummary($resourceMetadata, $documentation);
+            $this->extractResponses($swaggerMetadata, $documentation, $resourceMetadata);
+            $this->extractParameters($swaggerMetadata, $documentation);
+        }
+
+        return $documentation;
+    }
+
+    /**
+     * @param $documentation
+     * @param $resourceMetadata
+     */
+    protected function extractSummary(ResourceMetadata $resourceMetadata, SwaggerDocumentation $documentation)
+    {
+        if (null === $documentation->getSummary()) {
+            $documentation->setSummary(sprintf('Retrieves the collection of %s resources.', $resourceMetadata->getShortName()));
+        }
+    }
+
+    /**
+     * @param SwaggerMetadata $swaggerMetadata
+     * @param $documentation
+     * @param $resourceMetadata
+     */
+    protected function extractResponses(
+        SwaggerMetadata $swaggerMetadata,
+        SwaggerDocumentation $documentation,
+        ResourceMetadata $resourceMetadata)
+    {
+        if (empty($documentation->getResponses())) {
+            $responseDefinitionKey = $this->swaggerDefinitions->get($swaggerMetadata, false);
+            $response = [
+                '200' => [
+                    'description' => sprintf('%s collection response', $resourceMetadata->getShortName()),
+                    'schema' => [
+                        'type' => 'array',
+                        'items' => ['$ref' => sprintf('#/definitions/%s', $responseDefinitionKey)],
+                    ],
+                ],
+            ];
+            $documentation->setResponses($response);
+        }
+    }
+
+    /**
+     * @param SwaggerMetadata $swaggerMetadata
+     * @param $documentation
+     */
+    protected function extractParameters(SwaggerMetadata $swaggerMetadata, SwaggerDocumentation $documentation)
+    {
+        if (empty($documentation->getParameters())) {
```

Comment:
Prefer using a guard clause (invert the test return early to reduce the level of indentation and the complexity of the block).